crates/ide/src/highlight_related.rs

```diff
@@ -0,0 +1,452 @@
+use hir::Semantics;
+use ide_db::{
+    base_db::FilePosition,
+    defs::Definition,
+    helpers::pick_best_token,
+    search::{FileReference, ReferenceAccess, SearchScope},
+    RootDatabase,
+};
+use syntax::{
+    ast, match_ast, AstNode,
+    SyntaxKind::{ASYNC_KW, AWAIT_KW, QUESTION, RETURN_KW, THIN_ARROW},
+    SyntaxNode, SyntaxToken, TextRange, WalkEvent,
+};
+
+use crate::{display::TryToNav, references, NavigationTarget};
+
+pub struct DocumentHighlight {
```

Comment:
Less than ideal name, as we don’t use document terminology in ide. I guess just RelatedRange or HighlightedRange would work?